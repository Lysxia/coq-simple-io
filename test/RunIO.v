From Coq Require Import
     String.
From ExtLib Require Import
     Monad.
From SimpleIO Require Import
     IO_Random
     SimpleIO.
Import
  MonadNotation.
Open Scope monad_scope.
Open Scope string_scope.

Definition coin : IO unit :=
  b <- ORandom.bool tt;;
  print_endline (if b : bool then "head" else "tail").

RunIOPackage "cppo".
RunIOPackage "zarith".

RunIO (ORandom.self_init tt;; coin;; coin;; coin).
