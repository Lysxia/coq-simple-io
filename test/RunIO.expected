Running (ORandom.init (int_of_nat 42);; coin;; coin;; coin) ...
tail
head
tail
